nixos/tests/nvidia-container-toolkit.nix

```diff
@@ -0,0 +1,158 @@
+import ./make-test-python.nix (
+  {
+    pkgs,
+    lib,
+    system,
+    ...
+  }:
+  let
+    unfreeAndInsecurePkgs = import ../.. {
+      inherit system;
+      config = {
+        allowUnfree = true;
+        permittedInsecurePackages = [ "openssl-1.1.1w" ];
+      };
+    };
+    testContainerImage =
+      let
+        testCDIScript = pkgs.writeShellScriptBin "test-cdi" ''
+          die() {
+            echo "$1"
+            exit 1
+          }
+
+          check_file_referential_integrity() {
+            echo "checking $file referential integrity"
+            files=$(set -o pipefail && \
+                    ${pkgs.glibc.bin}/bin/ldd "$1" | \
+                    ${pkgs.gnugrep}/bin/grep '=>' | \
+                    ${pkgs.gnused}/bin/sed "s/.* => //" | \
+                    ${pkgs.gnused}/bin/sed "s/ (.*//") || exit 1
+
+            for file in $files; do
+              if [ ! -f "$file" ]; then
+                die "$file does not exist in the container filesystem"
+              fi
+            done
+          }
+
+          check_directory_referential_integrity() {
+            ${pkgs.findutils}/bin/find "$1" -type f -print0 | while read -d $'\0' file; do
```

Comment:
I usually do the `while ... done < <( subprocess without a subshell )` pattern, although I've been told some people prefer the subshell for whatever reason

Reply:
No special reason, let me know if you think I should change it, but I am fine with both.